src/api.proposed.notebookEnvironment.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { Event, Uri } from 'vscode';
+
+declare module './api' {
+    /**
+     * These types are not required for any other extension, except for the Python extension.
+     * Hence the reason to keep this separate. This way we can keep the API stable for other extensions (which would be the majority case).
+     */
+    export interface Jupyter {
+        /**
+         * This event is triggered when the environment associated with a Jupyter Notebook or Interactive Window changes.
+         * The Uri in the event is the Uri of the Notebook/IW.
+         */
+        onDidChangePythonEnvironment: Event<Uri>;
+        /**
+         * Returns the EnvironmentPath to the Python environment associated with a Jupyter Notebook or Interactive Window.
+         * If the Uri is not associated with a Jupyter Notebook or Interactive Window, then this method returns undefined.
+         * @param uri
+         */
+        getPythonEnvironment(uri: Uri):
+            | undefined
+            | {
```

Comment:
I would really just return the python executable path. Since we intend to switch to a different style of API. we can either wait here for the same of this API to be consistent with python environments extension, or return path which can be used to get the details from python environment extension.

Reply:
> Since we intend to switch to a different style of API. we can either wait here for the same of this API to be consistent with python environments extension

Lets wait, i'd prefer to return something thats consistent with what we have today.
Else its a mish mash of different things.